tests/net/lib/lwm2m/content_oma_tlv/src/main.c

```diff
@@ -1083,84 +1143,15 @@ static void test_get_objlnk(void)
 	}
 }
 
-static void test_get_objlnk_nodata(void)
+ZTEST(net_content_oma_tlv, test_get_objlnk_nodata)
 {
 	int ret;
 	struct lwm2m_objlnk value;
 
+	test_prepare_nodata();
+
 	ret = oma_tlv_reader.get_objlnk(&test_in, &value);
 	zassert_equal(ret, -ENODATA, "Invalid error code returned");
 }
 
-void test_main(void)
-{
-	ztest_test_suite(
-		lwm2m_content_plain_text,
-		ztest_unit_test_setup_teardown(
-			test_put_s8, test_prepare, unit_test_noop),
-		ztest_unit_test_setup_teardown(
-			test_put_s8_nomem, test_prepare_nomem, unit_test_noop),
```

Comment:
I agree with @enjiamai. These should be separated as two suites with two different before. They'll both run since they're both in the same binary.

Reply:
OK, updated. Thanks to review